test/e2e/framework/framework.go

```diff
@@ -226,3 +251,30 @@ func (f *Framework) AfterEach() {
 		Failf(strings.Join(messages, ","))
 	}
 }
+
+// createCloudProviderConfig unmarshalls the CCM's cloud provider config from the specified location so it can be used for testing.
+func createCloudProviderConfig(cloudConfigFile string) (*oci.Config, error) {
+	file, err := os.Open(cloudConfigFile)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("Couldn't open cloud provider configuration: %s.", cloudConfigFile))
+	}
+	defer file.Close()
+	cloudProviderConfig, err := oci.ReadConfig(file)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("Couldn't create cloud provider configuration: %s.", cloudConfigFile))
+	}
+	return cloudProviderConfig, nil
+}
+
+// createOCIClient creates an OCI client derived from the CCM's cloud provider config file.
+func createOCIClient(cloudProviderConfig *oci.Config) (client.Interface, error) {
```

Comment:
I did not see the point in adding configuration files. We grab the auth info from the cloud config.